sdk/identity/azure_identity/README.md

```diff
@@ -1,77 +1,425 @@
+
+This allows for trying all of the developer credentials on your machine while having predictable deployed behavior.
+
+## Examples
+
+The following examples are provided:
+<!-- no toc -->
+* [Authenticate with DefaultAzureCredential](#authenticate-with-defaultazurecredential "Authenticate with DefaultAzureCredential")
+<!-- TODO: Un-Comment if/when the sections are added>
+* [Define a custom authentication flow with ChainedTokenCredential](#define-a-custom-authentication-flow-with-chainedtokencredential "Define a custom authentication flow with ChainedTokenCredential")
+* [Async credentials](#async-credentials "Async credentials")
+</!-->
+
+### Authenticate with `DefaultAzureCredential`
+
+More details on configuring your environment to use `DefaultAzureCredential` can be found in the class's [reference documentation][default_cred_ref].
+
+This example demonstrates authenticating the `CosmosClient` from the [azure_data_cosmos][azure_data_cosmos] library using `DefaultAzureCredential`.
 
 ```rust
-use azure_core::credentials::TokenCredential;
+use azure_data_cosmos::CosmosClient;
 use azure_identity::DefaultAzureCredential;
-# use azure_core::{ClientOptions, Result};
-# use std::sync::Arc;
-# struct StorageAccountClient;
-# impl StorageAccountClient {
-#     fn new(_endpoint: &str, _credential: Arc<dyn TokenCredential>, _options: Option<ClientOptions>) -> Result<Arc<Self>> {
-#         Ok(Arc::new(StorageAccountClient))
-#     }
-# }
-# struct SecretClient;
-# impl SecretClient {
-#     fn new(_endpoint: &str, _credential: Arc<dyn TokenCredential>, _options: Option<ClientOptions>) -> Result<Arc<Self>> {
-#         Ok(Arc::new(SecretClient))
-#     }
-# }
 
 let credential = DefaultAzureCredential::new().unwrap();
-let storage_client = StorageAccountClient::new(
-    "https://myaccount.blob.storage.azure.net",
-    credential.clone(),
-    None,
-);
-let secret_client = SecretClient::new("https://myvault.keyvault.azure.net",
-    credential.clone(),
-    None,
-);
+let client = CosmosClient::new("https://myaccount.documents.azure.com/", credential, None).unwrap();
+```
+
+<!-- TODO: update this section on enabling interactive auth if/when it's configurable in the Rust library>
```

Comment:
Let's get rid of all this commented code. It serves no purpose and we can always copy it later. It's not uncommon for developers to read the raw markdown. When I'm working in a repo and have it cloned locally, I do.

Reply:
It's gone.